### DISCUSSION TITLE
resteasy-reactive multipart and swagger upload file

### DISCUSSION THREAD

**Discussion by MikhailMalyutin** [quarkusio/quarkus#31807] [Q&A] [answered]
Hi.

Looks like mutiny api is deprecated and quarkus suggest to migrate to resteasy-reactive.

But, after migration i catch some problems.

Before migration to reactive form part was:
```   
@PartType(MediaType.APPLICATION_OCTET_STREAM)
@field:Schema(type= SchemaType.STRING, description = "input", format = "binary")
var fileStream: InputStrean? = null
```

But in this case looks like APPLICATION_OCTET_STREAM is ignored and file is uploaded as text.

I read recommendations from https://quarkus.io/guides/resteasy-reactive#handling-multipart-form-data and see now it is recommended to use File or FileUpload. But if i use code like:
```
@RestForm("file")
@PartType(MediaType.APPLICATION_OCTET_STREAM)
@field:Schema(type= SchemaType.STRING, description = "input", format = "binary")
var fileStream: File? = null
```
then swagger don't see Schema and upload button is not displayed.

Are some standart way to have needed functionality?

I found some workaround like https://dev.to/felipewind/uploading-a-file-through-swagger-in-quarkus-3l8l but this solution with manual describing schema is not elegant.

[a comment by an automated account is not shown]

**Comment by MikeEdgar** [comment]
@MikhailMalyutin do you have a small reproducer app that we can look at?

**Reply by MikhailMalyutin** [reply]
https://github.com/MikhailMalyutin/bug-reproducer

**Comment by MikeEdgar** [comment, marked as answer]
@MikhailMalyutin , please try adding `implementation = String::class` to your `@field:Schema` on the `File` property. Until this is handled automatically in smallrye-open-api, that hint will set the correct `type` on the schema.

**Reply by MikhailMalyutin** [reply]
Thanks, all works fine!

**Comment by kps2024** [comment]
Thanks. It is working. I tried like

    @RestForm("profile_picture")
    @PartType(MediaType.APPLICATION_OCTET_STREAM)
    @Schema(type = SchemaType.STRING, format = "binary", description = "Uploaded image file", implementation = String.class)
    public FileUpload profilePicture;
